callgraph/core.py

```diff
@@ -43,13 +46,23 @@ def __init__(self, name):
         self.is_last_node = False
         self.code = []
         self.loc = 0
+        self.node_width = 0
+        self.node_height = 0
 
     def AddConnection(self, dst, kind, line_number=NO_LINE_NUMBER):
         self.connections.add(Connection(dst, kind, line_number))
+
+    def UpdateWidth(self):
+        self.node_width = self.loc * NODE_WIDTH_FACTOR

```

Comment:
On large files (100s of nodes, some nodes >100 lines of code), the resulting output cannot be converted in PNG because `dot` complains that the Cairo renderer goes out of memory.

I guess that instead of just linearly increasing the sizes of the node you could choose a given max size (e.g., 10 times the default?) and then have the largest node be 10 times bigger than the smallest one, with all the others taking sizes in between in proportion to the size.

Reply:
Max size of a node is regulated by its width which cannot exceed by five times the node width factor. Height is computed as half the width.